lib/matplotlib/axes/_axes.py

```diff
@@ -3675,8 +3675,7 @@ def dopatch(xs, ys, **kwargs):
 
         # width
         if widths is None:
-            distance = max(positions) - min(positions)
-            widths = [min(0.15 * max(distance, 1.0), 0.5)] * N
+            widths = [np.clip(np.ptp(positions), 0.15, 0.5)] * N
```

Comment:
Good catch.  Fixed, and updated the docstring to reflect the actual behavior (which was misdocumented).